### PR TITLE
feat(web2): add IPv6 configuration tab

### PR DESCRIPTION
**Pull request by marcellorinaldo**
This PR adds a net tab "IPv6" to the networking section that allows to configure IPv6-related settings.

**Related Issue:** N/A.

**Description of the solution adopted:** Summary of changes:
- Added UI tab
- Added validations
- Refactor of the network tabs orcherstrator code
- Backend code not implemented in this PR, will be separate

**Screenshots:**

Screenshot 2023-08-10 at 12 08 29

**Manual Tests**:
- [x] *Address Generation Mode* and *Privacy* are enabled only when *Configure = SLAAC*
- [x] When *Status = Enabled for LAN* and *Configure = Manually* then only *IP address* and *Subnet Mask* are editable.
- [x] When *Status = Enabled for WAN* and *Configure = Manually* then *WAN Priority*, *IP address*, *Subnet Mask*, *Gateway*, and *DNS Servers* are editable.
- [x] When *Status = Enabled for WAN* and *Configure != Manually* the *DNS Servers* is editable
- [x] *WAN Priority* is enabled only when *Status = Enabled for WAN*
- [x] All fields are disabled when *Status = Disabled* or a Loopback interface is selected
- [x] Tabs order (IPv4, IPv6, Wireless, DHCPv4&NAT, Modem, etc.) is the same as prior this PR
- [x] *WAN Priority* accepts only integers > -1
- [x] *Subnet mask* accepts only integers between 1 and 128
- [x] Fields that accept IPv6 addresses have working validation, even with shortened addresses. Some test cases can be found below
- [x] IPv6 tab is shown only on generic profiles
- [x] *Information* hints are correctly shown when hovering over the enabled elements

IPv6 addresses (positive) test cases:
- `2001:0000:3238:DFE1:0063:0000:0000:FEFB`
- `2001:0000:3238:DFE1:63:0000:0000:FEFB`
- `2001:0000:3238:DFE1:63::FEFB`
- `2001:0:3238:DFE1:63::FEFB`
- `::`
- `::1`
- `FE80::`

**Any side note on the changes made:** N/A.

### CODE REVIEW

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java [hunk 1]
```diff
@@ -245,73 +247,88 @@ public void setNetInterface(GwtNetInterfaceConfig selection) {
...
             showWirelessTabs();
-            if (!this.isWirelessAP) {
+            if (!isWirelessAP) {
                 includeDhcpNat = false;
             }
-        } else if (this.netIfConfig instanceof GwtModemInterfaceConfig) {
+        } else if (wrapper.isModem()) {
             includeDhcpNat = false;
-            this.modemGpsTabAnchorItem.setEnabled(this.isGpsSupported && !this.isIpv4Unmanaged);
+            this.modemGpsTabAnchorItem.setEnabled(wrapper.isGpsSupported() && !isIpv4Unmanaged);
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java:305]: The behavior here (as the same of the previous one) is not to show the gps tab if the modem is unmanaged. Is it the expected behavior? Probably this is out of the scope of this PR, but it worth to discuss on it.

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties [hunk 2]
```diff
@@ -315,9 +315,8 @@ netIPv4Address=IP Address
 netIPv4SubnetMask=Subnet Mask
 netIPv4Gateway=Gateway
 netIPv4DNSServers=DNS Servers
-netIPv4SearchDomains=Search Domains
 netIPv4InvalidPriority=Priority allowed values are only integers from -1 to 2147483647.
-netIPv4InvalidAddress=Invalid IPv4 Address
+netIPv4InvalidAddress=Invalid IP Address
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties]: Is this used also for IPv6? If so, maybe it is better to add a new message entry...

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages_ja.properties [hunk 3]
```diff
@@ -327,6 +326,23 @@ netConfigChangeConfirm=ネットワーク環境設定を変更しようとして
...
+
+netIPv6ToolTipAutoconfiguration=RFC4862 IPv6 Stateless Address Autoconfiguration で使用するアドレスを作成するための設定方法<br><br><b>EUI64</b><br>アドレスは、ハードウェア アドレスから派生したインターフェイス トークンを使用して生成されます。 これにより、アドレスのホスト部分が一定になり、ネットワークが変更されたときにホストの存在を追跡できるようになります。<br><br><b>Stable-privacy</b><br>RFC7217 で指定されている接続の安定 ID およびネットワーク アドレスとともに、ホスト固有の秘密キーの暗号的に安全なハッシュの使用を有効にします。 これにより、アドレス トラック ホストのプレゼンスを使用できなくなります。
+
+netIPv6ToolTipAddress=IPv6 アドレスの範囲は、0000:0000:0000:0000:0000:0000:0000:0000 から ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff です。 短縮ルールを適用できます。<br><br><b>先頭のゼロを省略します</b><br>2001:0000:3238:DFE1:0063:0000:0000:FEFB は 2001:0000:3238:DFE1:63:0000:0000:FEFB と記述できます (5 番目のブロック)<br><br><b>連続した 0 のブロックを置き換えます</b><br>2001:0000:3238:DFE1:63:0000:0000:FEFB can be written as 2001:0000:3238:DFE1:63::FEFB (6th and 7th blocks)<br><br><b>0 のブロックを短くする</b><br>2001:0000:3238:DFE1:63::FEFB は 2001:0:3238:DFE1:63::FEFB として記述できます (2 番目のブロック)
+
+netIPv6ToolTipPrivacy=RFC4941 で説明されているように、SLAAC の IPv6 プライバシー拡張機能を構成します。 有効にすると、カーネルは、修正された EUI-64 を介して MAC アドレスから生成されるパブリック アドレスに加えて、一時的な IPv6 アドレスを生成します。 この設定は、IPv6 アドレスによるホスト追跡を回避する別の方法として [アドレス生成モード] プロパティで有効にできる安定したプライバシー アドレスとは異なることに注意してください。
+
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages_ja.properties:361]: Have you check this with a native speaker?

**Review by pierantoniomerlino — CHANGES_REQUESTED** [review]

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties (hunk 2 above)
Comment (marcellorinaldo) [reply kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties]: I didn't want to replicate the same message twice for IPv4 and IPv6

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java (hunk 1 above)
Comment (marcellorinaldo) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java:305]: I agree we should discuss it further but outside this PR.

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties (hunk 2 above)
Comment (marcellorinaldo) [reply kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties]: Latest commit separates better the two cases.

**Comment by sfiorani** [comment]
Tested on Raspberry Pi, with `generic-aarch64` generic profile: everything is ok

**Review by pierantoniomerlino — APPROVED** [review]

**Pull request merged**